helpers/Helpers.ts

```diff
@@ -331,3 +331,412 @@ export const formatNumber = (num: number): string => {
+    minTdh: 340000,
+    level: 41
+  },
+  {
+    minTdh: 360000,
+    level: 42
+  },
+  {
+    minTdh: 380000,
+    level: 43
+  },
+  {
+    minTdh: 400000,
+    level: 44
+  },
+  {
+    minTdh: 420000,
+    level: 45
+  },
+  {
+    minTdh: 440000,
+    level: 46
+  },
+  {
+    minTdh: 460000,
+    level: 47
+  },
+  {
+    minTdh: 480000,
+    level: 48
+  },
+  {
+    minTdh: 500000,
+    level: 49
+  },
+  {
+    minTdh: 550000,
+    level: 50
+  },
+  {
+    minTdh: 600000,
+    level: 51
+  },
+  {
+    minTdh: 650000,
+    level: 52
+  },
+  {
+    minTdh: 700000,
+    level: 53
+  },
+  {
+    minTdh: 750000,
+    level: 54
+  },
+  {
+    minTdh: 800000,
+    level: 55
+  },
+  {
+    minTdh: 850000,
+    level: 56
+  },
+  {
+    minTdh: 900000,
+    level: 57
+  },
+  {
+    minTdh: 950000,
+    level: 58
+  },
+  {
+    minTdh: 1000000,
+    level: 59
+  },
+  {
+    minTdh: 1250000,
+    level: 60
+  },
+  {
+    minTdh: 1500000,
+    level: 61
+  },
+  {
+    minTdh: 1750000,
+    level: 62
+  },
+  {
+    minTdh: 2000000,
+    level: 63
+  },
+  {
+    minTdh: 2250000,
+    level: 64
+  },
+  {
+    minTdh: 2500000,
+    level: 65
+  },
+  {
+    minTdh: 2750000,
+    level: 66
+  },
+  {
+    minTdh: 3000000,
+    level: 67
+  },
+  {
+    minTdh: 3250000,
+    level: 68
+  },
+  {
+    minTdh: 3500000,
+    level: 69
+  },
+  {
+    minTdh: 3750000,
+    level: 70
+  },
+  {
+    minTdh: 4000000,
+    level: 71
+  },
+  {
+    minTdh: 4250000,
+    level: 72
+  },
+  {
+    minTdh: 4500000,
+    level: 73
+  },
+  {
+    minTdh: 4750000,
+    level: 74
+  },
+  {
+    minTdh: 5000000,
+    level: 75
+  },
+  {
+    minTdh: 5500000,
+    level: 76
+  },
+  {
+    minTdh: 6000000,
+    level: 77
+  },
+  {
+    minTdh: 6500000,
+    level: 78
+  },
+  {
+    minTdh: 7000000,
+    level: 79
+  },
+  {
+    minTdh: 7500000,
+    level: 80
+  },
+  {
+    minTdh: 8000000,
+    level: 81
+  },
+  {
+    minTdh: 8500000,
+    level: 82
+  },
+  {
+    minTdh: 9000000,
+    level: 83
+  },
+  {
+    minTdh: 9500000,
+    level: 84
+  },
+  {
+    minTdh: 10000000,
+    level: 85
+  },
+  {
+    minTdh: 11000000,
+    level: 86
+  },
+  {
+    minTdh: 12000000,
+    level: 87
+  },
+  {
+    minTdh: 13000000,
+    level: 88
+  },
+  {
+    minTdh: 14000000,
+    level: 89
+  },
+  {
+    minTdh: 15000000,
+    level: 90
+  },
+  {
+    minTdh: 16000000,
+    level: 91
+  },
+  {
+    minTdh: 17000000,
+    level: 92
+  },
+  {
+    minTdh: 18000000,
+    level: 93
+  },
+  {
+    minTdh: 19000000,
+    level: 94
+  },
+  {
+    minTdh: 20000000,
+    level: 95
+  },
+  {
+    minTdh: 21000000,
+    level: 96
+  },
+  {
+    minTdh: 22000000,
+    level: 97
+  },
+  {
+    minTdh: 23000000,
+    level: 98
+  },
+  {
+    minTdh: 24000000,
+    level: 99
+  },
+  {
+    minTdh: 25000000,
+    level: 100
+  }
+].reverse();
+
+export function tdh2Level(tdh: number): number {
+  return LEVELS.find((l) => tdh >= l.minTdh)?.level ?? 0;
+}
```

Comment:
can we move this in its own file or something? or the LEVELS array maybe and keep the function in helpers?